edgecast/client/client.go

```diff
@@ -132,6 +133,82 @@ func (c Client) BuildRequest(
 	return req, nil
 }
 
+// BuildRequest creates a new Request for the Edgecast API,
+// adding appropriate headers
+func (c Client) PrepareRequest(
+	method, path string,
+	body interface{},
+	queryParams map[string]string,
+	pathParams map[string]string,
+) (*retryablehttp.Request, error) {
+
+	// Adding Path Params
+	for k, v := range pathParams {
+		path = strings.Replace(path, "{"+k+"}", fmt.Sprintf("%v", v), -1)
+	}
+
+	relativeURL, err := url.Parse(path)
+	if err != nil {
+		return nil, fmt.Errorf("Client.BuildRequest: url.Parse: %v", err)
+	}
+
+	absoluteURL := c.Config.BaseAPIURL.ResolveReference(relativeURL)
+
+	// Adding Query Param
+	query := absoluteURL.Query()
+	for k, v := range queryParams {
+		query.Add(k, v)
+	}
+	// Encode the parameters.
+	absoluteURL.RawQuery = query.Encode()
+
+	var payload interface{}
+
+	if body != nil {
+		switch b := body.(type) {
+		case string:
+			payload = []byte(b)
+		default:
+			buf := new(bytes.Buffer)
+			err := json.NewEncoder(buf).Encode(body)
+			if err != nil {
+				return nil, err
+			}
+			logMsg := jsonutil.CreateRequestBodyLogMessage(
+				method,
+				absoluteURL.String(),
+				body)
+			c.Config.Logger.Debug(logMsg)
+			payload = buf
+
+		}
+	}
+
+	req, err := retryablehttp.NewRequest(method, absoluteURL.String(), payload)
+
+	if err != nil {
+		return nil, fmt.Errorf("BuildRequest: NewRequest: %v", err)
+	}
+
+	if body != nil {
```

Comment:
If body is a string type, should it be "text/html"?

Reply:
Updated to add content type as "text/plain; charset=utf-8", when bodytype is string

Reply:
Updated to add content type "text/plain; charset=utf-8" when body is of type string